xml/snmp_sysdescr.xml

```diff
@@ -2322,6 +2322,21 @@ Copyright (c) 1995-2005 by Cisco Systems
     <param pos="1" name="os.version"/>
   </fingerprint>
 
+  <!--======================================================================
+                            Fortinet FortiGate
+   =======================================================================-->
+
+  <fingerprint pattern="^[\w\-]+\s+v([\d\.]+),build\d+,\d+(?:\s+\(\w+\))?$">
```

Comment:
Given the examples presented is there any reason not to tighten this regex up with a strict match for `FortiaGate-` at the beginning?

Reply:
In addition, it appears the values that follow the dash are a model number. Looking at the [Fortinet Product Matrix](https://www.fortinet.com/content/dam/fortinet/assets/data-sheets/Fortinet_Product_Matrix.pdf) these are 2-4 digits followed by a single uppercase letter. The regex could match the model number and capture the value into the `hw.model` parameter, which would be positioned above `hw.device`.